processor/groupbytraceprocessor/event_test.go

```diff
@@ -498,21 +499,31 @@ func TestForceShutdown(t *testing.T) {
 	time.Sleep(100 * time.Millisecond)
 }
 
-func TestDoWithTimeout(t *testing.T) {
+func TestDoWithTimeout_NoTimeout(t *testing.T) {
 	// prepare
-	start := time.Now()
+	wantErr := errors.New("my error")
+	// test
+	succeed, err := doWithTimeout(20*time.Millisecond, func() error {
+		return wantErr
+	})
+	assert.True(t, succeed)
+	assert.Equal(t, wantErr, err)
+}
 
-	done := make(chan struct{})
+func TestDoWithTimeout_TimeoutTrigger(t *testing.T) {
+	// prepare
+	start := time.Now()
 
 	// test
-	doWithTimeout(5*time.Millisecond, func() error {
-		<-done
+	succeed, err := doWithTimeout(20*time.Millisecond, func() error {
+		<-time.After(1 * time.Second)
```

Comment:
Is this any different than time.Sleep(1 * time.Second)

Reply:
No :)